smm/lotus_adapter.go

```diff
@@ -0,0 +1,253 @@
+package smm
+
+import (
+    "context"
+    "fmt"
+    "github.com/filecoin-project/lotus/api"
+    "github.com/filecoin-project/lotus/build"
+    "github.com/filecoin-project/lotus/chain/actors"
+    laddress "github.com/filecoin-project/lotus/chain/address"
+    "github.com/filecoin-project/lotus/chain/store"
+    "github.com/filecoin-project/lotus/chain/types"
+    "github.com/ipfs/go-cid"
+    typegen "github.com/whyrusleeping/cbor-gen"
+)
+
+type lotusAdapter struct {
+    actor        laddress.Address
+    miner        laddress.Address
+    fullNode     api.FullNode
+    listener     StateChangeHandler
+    headChanges  <-chan[]*store.HeadChange
+    context      context.Context
+}
+
+func NewNode(ctx context.Context, fullNode api.FullNode, actor, miner Address, listener StateChangeHandler) (Node, *StateChange, error) {
```

Comment:
FullNode is a very large interface, way more than the adapter needs. I can see this caused the tests to require a lot of set-up boilerplate, with integration-style tests being the only possibility.

Re-declare a minimal interface subset here and consume that. Then you'll be able to write much more focussed tests, that test only the adapter itself, with a fake implementation.